### PR TITLE
EES-1125 Add dataLastPublished field to determine whether cached resp…

### PR DESCRIPTION
**Pull request by shakes90**
…onse should be used.

### CODE REVIEW

File: src/explore-education-statistics-common/src/modules/find-statistics/hooks/useTableQuery.ts [hunk 1]
```diff
@@ -32,18 +40,28 @@ export default function useTableQuery(
...
       .catch(() => null);
 
-    if (!response) {
-      response = await tableBuilderService.getTableData(query, releaseId);
+    const dataExpired =
+      item != null &&
+      dataLastPublished !== '' &&
+      new Date(dataLastPublished) > new Date(item.dataLastPublished);
+
+    if (item == null || dataExpired) {
```
Comment (ntsim) [inline src/explore-education-statistics-common/src/modules/find-statistics/hooks/useTableQuery.ts:52]: Think this proposed invalidation technique is a bit overkill. If i'm not mistaken, all we have to do is add `dataLastPublished` into the serialized `queryKey` and job done.

```typescript
const queryKey = JSON.stringify({
  ...query,
  dataLastPublished,
});
```

Comment (shakes90) [reply src/explore-education-statistics-common/src/modules/find-statistics/hooks/useTableQuery.ts:52]: good idea sir

Comment (shakes90) [reply src/explore-education-statistics-common/src/modules/find-statistics/hooks/useTableQuery.ts]: Done - Thanks

File: src/explore-education-statistics-common/src/modules/find-statistics/hooks/useTableQuery.ts [hunk 2]
```diff
@@ -30,13 +31,16 @@ export default function useTableQuery(
...
+      ...query,
+      dataLastPublished,
+    });
 
     let response = await storageService
       .get<TableDataResponse>(queryKey)
       .catch(() => null);
 
-    if (!response) {
+    if (response == null) {
```
Comment (ntsim) [inline src/explore-education-statistics-common/src/modules/find-statistics/hooks/useTableQuery.ts:43]: Would keep this check as `!response`. Using `== null` is more specific than we need, and potentially introduces bugs where it passes even though the value is potentially invalid e.g. an empty string.

It's better to just lean on Javascript coercion rules in these instances.

Comment (shakes90) [reply src/explore-education-statistics-common/src/modules/find-statistics/hooks/useTableQuery.ts]: Done

Comment (shakes90) [reply src/explore-education-statistics-common/src/modules/find-statistics/hooks/useTableQuery.ts]: done

**Pull request merged**